### DISCUSSION TITLE
meson.build:2030:22: ERROR: Expecting eol got eol_cont if build_tests[0] and ^

### DISCUSSION THREAD

**Discussion by Marietto2008** [mesonbuild/meson#12199] [Q&A]
Hello.

I'm trying to configure and build libvirt from source on Debian Buster on top of my ARM Chromebook,but I've got an error that I'm not able to fix :

```
# git clone https://github.com/libvirt/libvirt.git
Cloning into 'libvirt'...

# cd libvirt

# meson setup build

The Meson build system 
Version: 0.49.2 
Source dir: /home/marietto/Dati/virt/libvirt 
Build dir: /home/marietto/Dati/virt/libvirt/build 
Build type: native build

meson.build:2030:22: ERROR:  Expecting eol got eol_cont. 
if build_tests[0] and ^
```

Someone can help ? thanks.

**Comment by eli-schwartz** [comment]
Debian Buster is very old, and has a version of meson from 2019. libvirt requires a newer meson version -- and also has syntax that results in a parser error on very old meson versions.

Normally, meson is supposed to print an error such as "Meson version is 0.49.2 but project requires >=0.56.0", but the parser error raises before that has a chance to be seen. I fixed this in 878c1604e6348f98602bbdd16d92fc63ed15ebea but only Meson 1.1.0 knows how to correctly report an error for build files that don't parse. :sob:  This is *terrible* UX but I don't think there's any way to fix this glaring problem in older versions.

If you enable buster-backports, it will provide a meson version new enough to build libvirt.

**Comment by Marietto2008** [comment]
Hello. Thanks for the answer. I can't enable backports,otherwise I can't install gtk3,that I want to enable when I configure qemu 5.1. So,I think that the better solution for this problem is to install meson from source. Is this correct ? And the version of meson to install depends on the version of python installed on the system. Right ?

**Reply by eli-schwartz** [reply]
You can install a newer meson version with pip or running using the ./meson.py script in the git repository root.

That being said it is probably possible to download a single package from the backports repo, and install just that package.
